bluemira/builders/EUDEMO/first_wall/divertor.py

```diff
@@ -0,0 +1,329 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+"""
+Define builder for divertor
+"""
+
+import enum
+import operator
+from typing import Any, Callable, Dict, Iterable, List, Sequence
+
+import numpy as np
+
+from bluemira.base.builder import BuildConfig, Builder, Component
+from bluemira.base.components import PhysicalComponent
+from bluemira.base.config import Configuration
+from bluemira.equilibria.equilibrium import Equilibrium
+from bluemira.equilibria.find import find_flux_surface_through_point, get_legs
+from bluemira.geometry.tools import find_point_along_wire_at_length, make_polygon
+from bluemira.geometry.wire import BluemiraWire
+
+
+class LegPosition(enum.Enum):
+    """
+    Enum classifying divertor/separatrix leg positions
+    """
+
+    INNER = enum.auto()
+    OUTER = enum.auto()
+
+
+def get_separatrix_legs(
+    equilibrium: Equilibrium,
+) -> Dict[LegPosition, List[BluemiraWire]]:
+    """
+    Find the separatrix legs for the given equilibrium.
+    """
+    # A flag specifying which end of the plasma (i.e., upper or lower)
+    # we want the legs from will need to be added
+    legs = get_legs(equilibrium)
+    separatrix_legs = {
+        LegPosition.INNER: [make_polygon(loop.xyz) for loop in legs["lower_inner"]],
+        LegPosition.OUTER: [make_polygon(loop.xyz) for loop in legs["lower_outer"]],
+    }
+    return separatrix_legs
+
+
+class DivertorBuilder(Builder):
+    """
+    Build an EUDEMO divertor.
+    """
+
+    COMPONENT_INNER_BAFFLE = "inner_baffle"
+    COMPONENT_OUTER_BAFFLE = "outer_baffle"
+    COMPONENT_DOME = "dome"
+    COMPONENT_INNER_TARGET = "inner_target"
+    COMPONENT_OUTER_TARGET = "outer_target"
+
+    _required_params = [
+        "div_L2D_ib",
+        "div_L2D_ob",
+        "div_Ltarg",
+        "div_open",
+    ]
+    _required_config: List[str] = []
+    _params: Configuration
+    _default_runmode: str = "mock"
```

Comment:
If we're not running a design problem then you shouldn't need to set the default run mode (and shouldn't need to add implementations for run / read /mock).

Reply:
I'd added those methods because I was getting `NotImplementedError`s. I see now that that's because the `runmode` for the `FirstWallBuilder` was being propagated to the divertor builder. 

It could be handled by just popping 'runmode` before we pass the config through. It feels like there should be a more automatic way. I'll have a think